File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())
            return false;
        int[] map1 = new int[26];
        int[] map2 = new int[26];
        for(char c : s.toCharArray())
            map1[c-'a']++;
        
        for(char c : t.toCharArray())
            map2[c-'a']++;

        for(int i = 0 ; i < 26 ; i++){
            if(map1[i] != map2[i])
                return false;
        }
        return true;
    }
}

File: 242-valid-anagram/valid-anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {
        if (s.length != t.length) return false
        val map = HashMap<Char,Int>()
        for(ele in s.toCharArray()){
            map[ele] = map.getOrDefault(ele,0)+1
        }
        for(ele in t.toCharArray()){
            if(!map.containsKey(ele)){
                return false
            }
            map[ele] = map[ele]!! - 1
            if( map[ele]== 0)
                map.remove(ele)
        }
        println(map.size)
        if(map.size > 0)
            return false
        return true
    }
}